kolibri/core/assets/src/api-resources/contentNode.js

```diff
@@ -45,21 +45,45 @@ class ContentNodeResource extends Resource {
     }
     const filteredResourceIds = this.filterAndCheckResourceIds(resourceIds);
     let promise;
-    this.detail_cache = this.detail_cache || {};
+    this.ancestor_cache = this.ancestor_cache || {};
     const key = this.cacheKey({ id }, filteredResourceIds);
-    if (!this.detail_cache[key]) {
+    if (!this.ancestor_cache[key]) {
       const url = this.urls[`${this.name}-ancestors`](
         ...this.resourceIds.map((resourceKey) => resourceIds[resourceKey]), id);
       promise = this.client({ path: url }).then(response => {
         if (Array.isArray(response.entity)) {
-          this.detail_cache[key] = response.entity;
+          this.ancestor_cache[key] = response.entity;
           return Promise.resolve(response.entity);
         }
         logging.debug('Data appears to be malformed', response.entity);
         return Promise.reject(response);
       });
     } else {
-      promise = Promise.resolve(this.detail_cache[key]);
+      promise = Promise.resolve(this.ancestor_cache[key]);
+    }
+    return promise;
+  }
+  fetchNextContent(id, resourceIds = {}) {
```

Comment:
This function is pretty dense/complex. I'm not able to parse it well enough to review it, but as long as you're confident in it working, looks good for now. But perhaps after we get this in, coming back and breaking it up a bit or at least commenting it well would be good.